Editorial suggestions (#191)
w3c/wai-wcag-em: procedure.md

Usually, this step involves the evaluation commissioner (who may or may not be the product's owner), to align expectations, and an initial exploration of the product.

> #### Step 1.1: Define the scope of the digital product  {#step1a}

Reviewer: @hidde I am not sure about this edit. I think leaving it as 'product' only slightly reduces the title and leaves the scope slightly open.

I will leave it for you though.

During this step the evaluator determines the minimum set of combinations of operating systems, web browsers, assistive technologies, and other user agents that the product is expected to work with, and that is in-line with the WCAG 2 guidance on accessibility support (linked above). This step is carried out in consultation with the evaluation commissioner to ensure common expectation for the targeted level of accessibility support. The product's owner and product's developer may also have such a list of combinations that the product was designed to support, which could be a starting point for this step. Depending on the purpose of the evaluation such a list may need to be updated. For example, to assess how well the product works with more current browsers.

> 

Author: @iadawn - further to our discussion yesterday, for grammatical reasons, a full stop should not be used if the following clause is not a complete sentence. The same applies to a semicolon.
Options are:

- Depending on the purpose of the evaluation such a list may need to be updated — for example, to assess how well the product works with more current browsers.
- Depending on the purpose of the evaluation such a list may need to be updated (for example, to assess how well the product works with more current browsers).
- Depending on the purpose of the evaluation such a list may need to be updated. An example of why a list may need to be updated is to assess how well the product works with more current browsers.

#### Step 2.3: Identify the variety of sample types {#step2c}

<p class="methodology-requirement"><strong id="req2c">Methodology Requirement 2.3:</strong> Identify the types of samples.</p>

Samples with varying styles, layouts, structures, and functionality often have varying support for accessibility. They are often generated by different templates and scripts, or authored by different people. They may appear differently, behave differently, and contain different content depending on the particular product user and context.

During this step the evaluator explores the target product to identify the different **types** of samples. The outcome of this step is a list of descriptions of the types of content identified, rather than specific instances of samples. This list will be used in the following steps to help select a representative sample set for evaluation.

##### Examples of sample types

Examples of the different types of samples that evaluators can look for include samples that:

*   vary in style, layout, structure, navigation, interaction, and visual design,
*   include different types of content, such as forms, tables, lists, headings, multimedia, and scripting,

> *   include different functional components, such as date pickers, lightboxes, and sliders,

Reviewer 2: actually 'lightboxes' is a very old example updating… 

```suggestion
*   include different functional components, such as date pickers, modal overlays, and carousels,
```

Particularly for new technologies it is not always possible to ensure that every accessibility feature provided on a digital product, such as a “Show captions” function in a media player, is supported by every possible combination of operating system, web browser, assistive technology, and other user agents. WCAG 2 does not pre-define which combinations of features and technologies must be supported as this depends on the particular context of the product, including its language, the technologies that are used to create the content, and the user agents currently available. [Understanding Accessibility Support](https://www.w3.org/WAI/WCAG22/Understanding/conformance#accessibility-support) provides more guidance on the WCAG 2 concept of _accessibility support_.

> During this step the evaluator determines the minimum set of combinations of operating systems, web browsers, assistive technologies, and other user agents that the product is expected to work with, and that is in-line with the WCAG 2 guidance on accessibility support (linked above). This step is carried out in consultation with the evaluation commissioner to ensure common expectation for the targeted level of accessibility support. The product's owner and product's developer may also have such a list of combinations that the product was designed to support, which could be a starting point for this step. Depending on the purpose of the evaluation such a list may need to be updated — for example to assess how well the product works with more current browsers.

Reviewer: ```suggestion
During this step the evaluator determines the minimum set of combinations of operating systems, web browsers, assistive technologies, and other user agents that the product is expected to work with, and that is in-line with the WCAG 2 guidance on accessibility support (linked above). This step is carried out in consultation with the evaluation commissioner to ensure common expectation for the targeted level of accessibility support. The product's owner and product's developer may also have such a list of combinations that the product was designed to support, which could be a starting point for this step. Depending on the purpose of the evaluation such a list may need to be updated. For example, to assess how well the product works with more current browsers.
```

> <p class="note">This initial definition of the baseline does not limit the evaluator from using additional operating systems, web browsers, assistive technologies and other user agents at a later point — for example to evaluate content that was not identified at this early stage of the evaluation process. In this case the baseline is extended with the additional tools that were used.</p>

Reviewer: ```suggestion
<p class="note">This initial definition of the baseline does not limit the evaluator from using additional operating systems, web browsers, assistive technologies and other user agents at a later point. For example, to evaluate content that was not identified at this early stage of the evaluation process. In this case the baseline is extended with the additional tools that were used.</p>
```

An evaluation commissioner may be interested in additional information beyond what is needed to evaluate the extent of conformance of the target product to WCAG 2. For example, an evaluation commissioner might be interested in:

*   evaluation of additional views beyond what is needed to form a representative sample set from the target digital product,
*   reports of all occurrences of issues rather than representative examples of the types of issues on the target digital product,
*   analysis of particular use cases, situations, and user groups for interacting with the target digital product,
*   description of possible solutions to the issues encountered beyond the scope of the evaluation,
*   evaluation involving users with disabilities, and
*   adherence to specific documentation or reporting templates.

> Such additional evaluation requirements that are agreed on with the evaluator need to be clarified early on and documented. This also needs to be reflected in the resulting report — for example, to clarify how the selection of the sample set was carried out.

Reviewer: ```suggestion
Such additional evaluation requirements that are agreed on with the evaluator need to be clarified early on and documented. This also needs to be reflected in the resulting report. For example, to clarify how the selection of the sample set was carried out.
```

*   evaluation involving users with disabilities, and
*   adherence to specific documentation or reporting templates.

Such additional evaluation requirements that are agreed on with the evaluator need to be clarified early on and documented. This also needs to be reflected in the resulting report. For example, to clarify how the selection of the sample set was carried out.

> 

Author: @iadawn For grammatical reasons, a full stop should not be used if the following clause is not a complete sentence. The same applies to a semicolon.
Options are:

- Such additional evaluation requirements that are agreed on with the evaluator need to be clarified early on and documented. This also needs to be reflected in the resulting report — for example, to clarify how the selection of the sample set was carried out.
- Such additional evaluation requirements that are agreed on with the evaluator need to be clarified early on and documented. This also needs to be reflected in the resulting report (for example, to clarify how the selection of the sample set was carried out).
- Such additional evaluation requirements that are agreed on with the evaluator need to be clarified early on and documented. This also needs to be reflected in the resulting report. An example of why this needs to be in the report is to clarify how the selection of the sample set was carried out.

Explore the target product to identify its common views, which may also be specific states of views. Typically these are linked directly from the main entry point of the target product (like the home page on a website, or the start screen of an app), and often linked from the header, navigation, and footer sections of other views. The outcome of this step is a list of all common pages or views of the target product.

> #### Step 2.2: Identify essential functionality of the digital product {#step2b}

Reviewer: @hidde I am not sure about this edit. As noted previously

<aside class="example" title="Product functionality examples">
<ul>

>     <li>selecting and purchasing products from the web shop,</li>

Reviewer: ```suggestion
    <li>Selecting and purchasing products from the web shop.</li>
```

This list is a standalone set of examples. Either have a lead in to make it a complete sentence (not my preference) or structure so that each item is a sufficiently complete sentence in and of itself.

>     <li>completing and submitting the survey forms, and</li>

Reviewer: ```suggestion
    <li>Completing and submitting the survey forms.</li>
```

>     <li>registering for an account on the product.</li>

Reviewer: ```suggestion
    <li>Registering for an account on the product.</li>
```

Samples with varying styles, layouts, structures, and functionality often have varying support for accessibility. They are often generated by different templates and scripts, or authored by different people. They may appear differently, behave differently, and contain different content depending on the particular product user and context.

During this step the evaluator explores the target product to identify the different **types** of samples. The outcome of this step is a list of descriptions of the types of content identified, rather than specific instances of samples. This list will be used in the following steps to help select a representative sample set for evaluation.

<aside class="example" title="Sample type examples">

> Examples of the different types of samples that evaluators can look for include samples that:

Reviewer: ```suggestion
Examples of the different types of sample that evaluators can look for include ones that:
```

Too many 'samples' and too many '...les'. Don't care how it is structured but less makes it more readable.

Also, these are examples of a sample. The plural of plural doesn't make sense to me.

Reviewer 2: Makes sense, thanks!

*   vary in style, layout, structure, navigation, interaction, and visual design,
*   include different types of content, such as forms, tables, lists, headings, multimedia, and scripting,

> *   include different functional components, such as date pickers, modal overlays, and carousel/sliders,

Reviewer: ```suggestion
*   include different functional components, such as date pickers, modal overlays, and carousels,
```
"Carousels and sliders", or either "carousels" or "sliders" not "carousels/sliders".

*   use different technologies, such as HTML, CSS, JavaScript, WAI-ARIA, and PDF,
*   are drawn from different areas of the product (such as home page, web shop, and other departments), including any applications,

> *   reflect different coding styles and [templates](#template) (if this is known to the evaluator),

Reviewer: ```suggestion
*   reflect different coding styles and [templates](https://www.w3.org/TR/ATAG20/#def-template) (if this is known to the evaluator),
```

<p class="methodology-requirement"><strong id="req2e">Methodology Requirement 2.5:</strong> Identify other samples that are relevant to people with disabilities and to accessibility of the digital product.</p>

Some digital products include samples that are specifically relevant for people with disabilities and the accessibility of the digital product. The outcome of this step is a list of such samples, if they have not already been identified as part of [Step 2.1: Identify common views of the digital product](#step2a).

##### Examples of other relevant samples

Examples of other relevant samples include those:

*   explaining the accessibility features of the digital product,
*   with information and help on the use of the digital product,
*   explaining settings, preferences, options, shortcuts, etc., and
*   with contact information, directions, and support instructions.

> ### Step 3: Select a representative sample set {#step3}

Reviewer 2: Ooh nice catch!

<p class="methodology-requirement"><strong id="req2e">Methodology Requirement 2.5:</strong> Identify other samples that are relevant to people with disabilities and to accessibility of the digital product.</p>

> Some digital products include samples that are specifically relevant for people with disabilities and the accessibility of the digital product. The outcome of this step is a list of such samples, if they have not already been identified as part of [Step 2.1: Identify common views of the digital product](#step2a).

Reviewer: @hidde If the choice is to not add 'digital' in the above points, then remove it from here too

> <aside class="example" title="Examples of other relevant samples include those:">

Reviewer: ```suggestion
<aside class="example" title="Other relevant samples">
<p>Other samples include those:</p>
```

Review how titles in the 'example' class are rendered in terms of HTML. It is better IMHO to have the title not be part of the content of the example but a coherent title in itself. Additionally, two ':' in one line is yeuch.

Reviewer 2: Great point, wasn't aware it renders like this, will double check.

*   **Complexity of the digital product** — higher complexity requires a larger sample set to evaluate; consider the following:
    *   **How interactive the content is** — products with content that is rich in interaction require larger sample sets to cover the functions provided by a sample and the different states that individual samples can have.
    *   **How the content is generated** — products with content that is aggregated from different sources or that is processed as it is served (at runtime) typically require larger sample sets to cover the combinations of content that can be generated.
    *   **How the content is implemented** — products that are available in different versions, served according to users and their preferences, or that adapt to access devices require larger sample sets to cover these different situations.
*   **Consistency of the product** — lower consistency requires a larger sample set to evaluate; consider the following:

>     *   **Variety of sample types** — products with a broader variety of sample types (see [Step 2.3: Identify the types of samples](#step2c)) require larger sample sets to evaluate.

Reviewer: ```suggestion
    *   ** Identify the variety of sample types** — products with a broader variety of sample types (see [Step 2.3: Identify the types of samples](#step2c)) require larger sample sets to evaluate.
```

The title is still 'Identify the variety of sample types'. If the change is necessary, change in both not one place.

    *   **Training for the developers** — products with designers, developers, and content authors who receive regular training tend to have more consistent accessibility performance, so they typically require smaller sample sets to evaluate.
    *   **Development tools being used** — products that are developed and maintained using a consistent set of tools such as a content management system (CMS) also tend to be more consistent and require smaller sample sets to evaluate.
    *   **Number of authors** — products that are developed and maintained by a more confined set of authors, including content editors, tend to be more consistent and require smaller sample sets to evaluate.
*   **Required level of confidence** — higher confidence in the evaluation results often requires evaluation of a larger sample set.
*   **Availability of prior evaluation findings** — smaller sample sets may be required when evaluators have access to prior evaluation findings, including test results from manual and automated accessibility testing.

> The selection carried out during this step relies initially on the exploration carried out in [Step 2: Explore the target digital product](#step2). The selection is also continually refined during the following [Step 4: Evaluate the selected sample set](#step4), as the evaluator learns more about the particular implementation aspects of the target product.

Reviewer: @hidde Change as appropriate based on thought about 'digital'.

2.  For each starting point for a process, identify and record at least the default sequence of samples to complete the process. Include these samples.  
    <p class="note">The default sequence follows the standard use case, describing the default path through the complete process. It assumes that there are no user input errors and no selection of additional options. For example, for a web shop application, the user would proceed to checkout, confirm the default payment option, provide all required payment details correctly, and complete the purchase, without changing the contents of the shopping cart, using a stored user profile, selecting alternative options for payment or shipping address, providing erroneous input, and so forth.</p>
3.  For each process, identify and record the branch sequences of samples that are commonly accessed and critical for the successful completion of the process. Include these samples.  
    <p class="note">Branch sequences may terminate where they re-enter the default branch of the process. For example, adding a new shipping address will be registered as a critical alternative branch that leads back to the default branch of the process.</p>

> <p class="note">In most cases it is necessary to record and specify the actions needed to proceed from one sample to the next in a sequence to complete a process so that they can be replicated later. An example of such action could be "fill out name and address, and select the 'Submit' button". In most cases the web address (URI) will not be sufficient to identify the sample in a complete process. It is also useful to clearly record when samples are part of a process so that evaluators can focus their effort on the relevant changes, such as elements that were added, modified, or made visible.</p>

Reviewer: ```suggestion
<p class="note">In most cases it is necessary to record and specify the actions needed to proceed from one sample to the next in a sequence to complete a process so that they can be replicated later. An example of such action could be "fill out name and address, and select the 'Submit' button". In most cases the web address (URL) will not be sufficient to identify the sample in a complete process. It is also useful to clearly record when samples are part of a process so that evaluators can focus their effort on the relevant changes, such as elements that were added, modified, or made visible.</p>
```

Most likely that a web address is a URL

#### Step 5.2: Record the evaluation specifics (optional) {#step5b}

<p class="methodology-requirement"><strong id="req5b">Methodology Requirement 5.2:</strong> Archive the samples evaluated, and record the evaluation tools, web browsers, assistive technologies, other software, and methods used to evaluate them (optional).</p>

> While optional, it is good practice for evaluators to keep record of the evaluation specifics — for example to support conflict resolution in the case of dispute. This includes archiving the samples evaluated, and recording the evaluation tools, web browsers, assistive technologies, other software, and methods used to evaluate them. This recording is typically kept internal and not shared by the evaluator unless otherwise agreed on in [Step 1.4: Define additional evaluation requirements (optional)](#step1d).

Reviewer: ```suggestion
While optional, it is good practice for evaluators to keep record of the evaluation specifics. For example, to support conflict resolution in the case of dispute. This includes archiving the samples evaluated, and recording the evaluation tools, web browsers, assistive technologies, other software, and methods used to evaluate them. This recording is typically kept internal and not shared by the evaluator unless otherwise agreed on in [Step 1.4: Define additional evaluation requirements (optional)](#step1d).
```

<p class="methodology-requirement"><strong id="req5b">Methodology Requirement 5.2:</strong> Archive the samples evaluated, and record the evaluation tools, web browsers, assistive technologies, other software, and methods used to evaluate them (optional).</p>

> While optional, it is good practice for evaluators to keep record of the evaluation specifics. For example, to support conflict resolution in the case of dispute. This includes archiving the samples evaluated, and recording the evaluation tools, web browsers, assistive technologies, other software, and methods used to evaluate them. This recording is typically kept internal and not shared by the evaluator unless otherwise agreed on in [Step 1.4: Define additional evaluation requirements (optional)](#step1d).

Author: @iadawn - some options to make the final clause grammatically correct:

- While optional, it is good practice for evaluators to keep record of the evaluation specifics — for example, to support conflict resolution in the case of dispute.
- While optional, it is good practice for evaluators to keep record of the evaluation specifics (for example, to support conflict resolution in the case of dispute).
- While optional, it is good practice for evaluators to keep record of the evaluation specifics. An example of why this is good practice is to support conflict resolution in the case of dispute.

**Reminder:** In the majority of situations, using this methodology alone does not result in [WCAG 2 conformance claims](https://www.w3.org/TR/WCAG22/#conformance-claims) for the target digital product; see [Relation to WCAG 2 Conformance Claims](#context) for more background.

Product owners may wish to make public statements about the outcomes from evaluations following this methodology. This can be done when at least every non-optional methodology requirement is satisfied, the conformance target defined in [Step 1.2: Define the conformance target](#step1b) is satisfied by all samples evaluated (in [Step 4: Evaluate the selected sample set](#step4)), and the product owner commits to ensuring the validity and maintaining the accuracy of the evaluation statement made.

An evaluation statement according to this methodology includes at least the following information:

1.  **Date** of when the evaluation statement was issued;
2.  **Guidelines title, version and URI:** "Web Content Accessibility Guidelines 2.2 at [https://www.w3.org/TR/WCAG22/](https://www.w3.org/TR/WCAG22/)";

> 3.  **Conformance level evaluated**: Level A, Level AA or Level AAA, as defined in [Step 1.2: Define the conformance target](#step1b);

Reviewer: ```suggestion
3.  **Conformance level evaluated**: Level A, AA or AAA, as defined in [Step 1.2: Define the conformance target](#step1b);
```
